### PR TITLE
feat: Changes for multitenancy dashboard

### PR DESCRIPTION
**Pull request by prateek3255**
## Summary of change

- Add a method to get plugin config properties

## Related issues
N/A

## Test Plan
N/A

## Documentation changes
N/A

## Checklist for important updates
- [x] Changelog has been updated
- [ ] `pluginInterfaceSupported.json` file has been updated (if needed)
- [ ] Changes to the version if needed
   - In `build.gradle`
- [x] Had installed and ran the pre-commit hook
- [ ] If there are new dependencies that have been added in `build.gradle`, please make sure to add them in `implementationDependencies.json`.
- [x] Issue this PR against the latest non released version branch.
   - To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
   - If no such branch exists, then create one from the latest released branch.

## Remaining TODOs for this PR
- [ ] Update the API to return all property values and more information about properties like if they are protected because of SaaS

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,8 @@ to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
 
+- Adds implementation for a new method `getConfigFieldsInfo` to fetch the plugin config fields.
+
```
Comment (rishabhpoddar) [inline CHANGELOG.md:11]: changes to schema?

File: src/main/java/io/supertokens/storage/postgresql/annotations/EditableInDashboard.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.FIELD)
+public @interface EditableInDashboard {
+}
```
Comment (rishabhpoddar) [inline src/main/java/io/supertokens/storage/postgresql/annotations/EditableInDashboard.java:27]: please explain this

File: src/main/java/io/supertokens/storage/postgresql/config/PostgreSQLConfig.java [hunk 3]
```diff
@@ -40,84 +43,109 @@ public class PostgreSQLConfig {
...
 
     @JsonProperty
     @UserPoolProperty
+    @ConfigDescription("Specify the port to use when connecting to PostgreSQL instance. (Default: 5432)")
+    @EditableInDashboard
     private int postgresql_port = -1;
 
     @JsonProperty
     @ConnectionPoolProperty
+    @ConfigDescription("The PostgreSQL user to use to query the database. If the relevant tables are not already created by you, this user should have the ability to create new tables. To see the tables needed, visit: https://supertokens.io/docs/community/getting-started/database-setup/postgresql")
```
Comment (rishabhpoddar) [inline src/main/java/io/supertokens/storage/postgresql/config/PostgreSQLConfig.java:68]: the link is incorrect

File: src/main/java/io/supertokens/storage/postgresql/config/PostgreSQLConfig.java [hunk 4]
```diff
@@ -40,84 +43,109 @@ public class PostgreSQLConfig {
...
     private String postgresql_user = null;
 
     @JsonProperty
     @ConnectionPoolProperty
+    @ConfigDescription("Password for the PostgreSQL user. If you have not set a password make this an empty string.")
```
Comment (rishabhpoddar) [inline src/main/java/io/supertokens/storage/postgresql/config/PostgreSQLConfig.java:73]: what is the default here?

File: CHANGELOG.md (hunk 1 above)
Comment (rishabhpoddar) [reply CHANGELOG.md:11]: + migration instra for them

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: src/main/java/io/supertokens/storage/postgresql/config/PostgreSQLConfig.java (hunk 3 above)
Comment (sattvikc) [reply src/main/java/io/supertokens/storage/postgresql/config/PostgreSQLConfig.java:68]: Fixed

File: src/main/java/io/supertokens/storage/postgresql/config/PostgreSQLConfig.java (hunk 4 above)
Comment (sattvikc) [reply src/main/java/io/supertokens/storage/postgresql/config/PostgreSQLConfig.java:73]: It's null. added default

File: src/main/java/io/supertokens/storage/postgresql/annotations/EditableInDashboard.java (hunk 2 above)
Comment (sattvikc) [reply src/main/java/io/supertokens/storage/postgresql/annotations/EditableInDashboard.java:27]: Added comment

File: CHANGELOG.md (hunk 1 above)
Comment (sattvikc) [reply CHANGELOG.md:11]: Release prep at the end

**Pull request merged**